### ISSUE TITLE
Wrong flow after Keycard freezing in logged-in state when opening "Change PIN/PUK/pairing code" screen

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#12328]
# Bug Report

## Problem 

If Keycard was frozen in logged-in state, e.g. when signing a transaction or changing some passcode, and User taps `Dismiss` in the pop-up and navigate to `Change PIN` (PUK / pairing) screen, the `Reset passcode with PUK` starts, even if the PUK has not been set previously.

#### Expected behavior

`Keycard is frozen` pop-up when opening any screen when the passcode is required

#### Actual behavior

https://user-images.githubusercontent.com/67952253/126484847-2eca72ec-9090-4dbe-ae1e-6fa293ecf18d.mp4

### Reproduction

1. Recover Keycard account
2. Settings > Keycard > Change 6-digit passcode
3. Enter invalid passcode 3 times (or freeze Keycard in any other way being logged-in)
4. Tap `Dismiss` in the pop-up
5. Tap back
6. Open `Change 12-digit PUK`

### Additional Information

- Status version: nightly 21/07
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by bitgamma** [comment]
I think we should re-think a bit this interaction, because now that we have backup cards, we probably shouldn't block any flow, because we don't know what card will be tapped after selecting an action.

my proposal is:

1. all menu, options etc assume you have a working/not blocked keycard so when you open them you get the intended functionality
2. if the tapped card is blocked, the pop-up is displayed as a reaction to this (this happens already)
3. we stop storing the "frozen" state completely and thus remove the corresponding banner from the wallet page

not completely sure about the last point since it is a nice and visible entry point to the reset flow, however since any flow will display the popup (including login) I don't know if it make sense to have a "frozen" state.

so long story short I would propose fixing this bug by having the change pin/puk/pairing always display the intended screen and only show the popup if a blocked card is tapped when verifying the PIN. Optionally, completely remove the frozen state.

wdyt?

/cc @guylouis
